### PR TITLE
[easy][cuda] Removed the warp size hardcode on layer norm backward CUDA kernel

### PR DESCRIPTION
**Pull request by valentinandrei**
Just a nit fix where `GammaBetaBackwardCUDAKernel_32x32` kernel used a hardcoded warp size for performing the reduction and laneId calculation. Changed this to use `C10_WARP_SIZE`.

### CODE REVIEW

**Comment by valentinandrei** [comment]
cc: @malfet

File: aten/src/ATen/native/cuda/layer_norm_kernel.cu [hunk 1]
```diff
@@ -568,7 +568,7 @@ __global__ void GammaBetaBackwardCUDAKernel_32x32(
     // Load transposed so that a warp holds an entire column
     T_ACC reg_dg = s_dg[threadIdx.x * padded_bx + threadIdx.y];
     T_ACC reg_db = s_db[threadIdx.x * padded_bx + threadIdx.y];
-    for (int delta = 16; delta >= 1; delta /= 2) {
+    for (int delta = C10_WARP_SIZE / 2; delta >= 1; delta /= 2) {
```
Comment (malfet) [inline aten/src/ATen/native/cuda/layer_norm_kernel.cu:571]: Nit (right shift if always faster than divide by 2)
```suggestion
    for (unsigned delta = C10_WARP_SIZE >> 1; delta >= 1; delta >>= 1) {
```

File: aten/src/ATen/native/cuda/layer_norm_kernel.cu [hunk 2]
```diff
@@ -504,7 +504,7 @@ __global__ void GammaBetaBackwardCUDAKernel_32x32(
 
   if (j < N) {
     constexpr int unroll_factor = 8;
-    int laneId = threadIdx.x & 0x1f;
+    int laneId = threadIdx.x % C10_WARP_SIZE;
```
Comment (malfet) [inline aten/src/ATen/native/cuda/layer_norm_kernel.cu:507]: Do you know if nvcc guarantees equivalence between `(foo & 0x1f)` and `(foo % 32)`? I think behavior is slightly different if `foo` is negative. Please consider, for consistency sake to use `& (size-1)`
```suggestion
    int laneId = threadIdx.x & (C10_WARP_SIZE -1);
```

**Review by malfet — APPROVED** [review]

**Pull request closed without merging**